crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -65,94 +70,148 @@ impl RpcHandler for ForkChoiceUpdatedV3 {
+                        latest_canonical_block_hash(&context.storage).unwrap(),
+                    ))
+                }
+                InvalidForkChoice::Syncing => {
+                    // Start sync
+                    let current_number = context.storage.get_latest_block_number()?.unwrap();
+                    let Some(current_head) =
+                        context.storage.get_canonical_block_hash(current_number)?
+                    else {
+                        return Err(RpcErr::Internal(
+                            "Missing latest canonical block".to_owned(),
+                        ));
+                    };
+                    let sync_head = forkchoice_state.head_block_hash;
+                    tokio::spawn(async move {
+                        // If we can't get hold of the syncer, then it means that there is an active sync in process
+                        if let Ok(mut syncer) = context.syncer.try_lock() {
+                            syncer
+                                .start_sync(current_head, sync_head, context.storage.clone())
+                                .await
+                        }
+                    });
+                    ForkChoiceResponse::from(PayloadStatus::syncing())
                 }
-                if attributes.timestamp <= head_block.timestamp {
-                    return Err(RpcErr::InvalidPayloadAttributes(
-                        "invalid timestamp".to_string(),
-                    ));
+                reason => {
+                    warn!("Invalid fork choice state. Reason: {:#?}", reason);
+                    return Err(RpcErr::InvalidForkChoiceState(reason.to_string()));
                 }
-                let args = BuildPayloadArgs {
-                    parent: self.fork_choice_state.head_block_hash,
-                    timestamp: attributes.timestamp,
-                    fee_recipient: attributes.suggested_fee_recipient,
-                    random: attributes.prev_randao,
-                    withdrawals: attributes.withdrawals.clone(),
-                    beacon_root: Some(attributes.parent_beacon_block_root),
-                    version: 3,
-                };
-                let payload_id = args.id();
-                response.set_id(payload_id);
-                let payload = match create_payload(&args, &context.storage) {
-                    Ok(payload) => payload,
-                    Err(ChainError::EvmError(error)) => return Err(error.into()),
-                    // Parent block is guaranteed to be present at this point,
-                    // so the only errors that may be returned are internal storage errors
-                    Err(error) => return Err(RpcErr::Internal(error.to_string())),
-                };
-                context.storage.add_payload(payload_id, payload)?;
-            }
+            };
+            Ok((None, forkchoice_response))
         }
+    }
+}
+
+fn build_payload_v3(
+    attributes: &PayloadAttributesV3,
```

Comment:
Here is where we could technically use a `PayloadAttributes` enum that contains all posibilities

```
pub enum PayloadAttributes {
    PayloadAttributesV1,
    PayloadAttributesV2,
    PayloadAttributesV3,
}
```